freqtrade/rpc/webhook.py

```diff
@@ -0,0 +1,68 @@
+"""
+This module manages webhook communication
+"""
+import logging
+from typing import Any,  Dict
+
+from requests import post, RequestException
+
+from freqtrade.rpc import RPC, RPCMessageType
+
+
+logger = logging.getLogger(__name__)
+
+logger.debug('Included module rpc.webhook ...')
+
+
+class Webhook(RPC):
+    """  This class handles all webhook communication """
+
+    def __init__(self, freqtrade) -> None:
+        """
+        Init the Webhook class, and init the super class RPC
+        :param freqtrade: Instance of a freqtrade bot
+        :return: None
+        """
+        super().__init__(freqtrade)
+
+        self._config = freqtrade.config
+
+    def cleanup(self) -> None:
+        """
+        Cleanup pending module resources.
+        This will do nothing for webhooks, they will simply not be called anymore
+        """
+        pass
+
+    def send_msg(self, msg: Dict[str, Any]) -> None:
+        """ Send a message to telegram channel """
+        try:
+
+            if msg['type'] == RPCMessageType.BUY_NOTIFICATION:
+                valuedict = self._config['webhook'].get('webhookbuy', None)
+            elif msg['type'] == RPCMessageType.SELL_NOTIFICATION:
+                valuedict = self._config['webhook'].get('webhooksell', None)
+            elif msg['type'] == RPCMessageType.STATUS_NOTIFICATION:
+                valuedict = self._config['webhook'].get('webhookstatus', None)
+            else:
+                raise NotImplementedError('Unknown message type: {}'.format(msg['type']))
+            if not valuedict:
```

Comment:
This condition can be removed, because if `valuedict` is None, the `NotImplementedError` will be raised a few lines above

Reply:
unfortunately i have to disagree, this can happen if `"webhookstatus"` is not configured, which is a valid case if i don't want to be notified via webhook for status updates.

Reply:
Ah I see, I assumed they are mandatory

Reply:
the "manual" reads (`webhook-config.md`) ... `Not all fields are necessary` :wink: